File: 2025/953. Verifying an Alien Dictionary/Solution.java
class Solution {
    public boolean isAlienSorted(String[] words, String order) {
        int[] map = new int[26];
        char[] arr = order.toCharArray();
        for (int i = 0; i < arr.length; i++) {
            map[arr[i] - 'a'] = i;
        }
        for (int i = 0; i < words.length - 1; i++) {
            if (!compare(words[i], words[i + 1], map))
                return false;
        }
        return true;
    }

    private boolean compare(String s1, String s2, int[] map) {
        for (int i = 0; i < s1.length(); i++) {
            if (i >= s2.length())
                return false;
            int compare = map[s1.charAt(i) - 'a'] - map[s2.charAt(i) - 'a'];
            if (compare < 0)
                return true;
            if (compare > 0)
                return false;
        }
        return true;
    }
}

File: 2025/953. Verifying an Alien Dictionary/solution.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        alien_dict = {item: i for i, item in enumerate(order)}
        for i in range(len(words) - 1):
            w1, w2 = words[i], words[i + 1]
            j = 0
            while j < len(w1) and j < len(w2):
                c, n = alien_dict[w1[j]], alien_dict[w2[j]]
                if c < n:
                    break;
                elif c > n or len(w2) <= j + 1 < len(w1):
                    return False
                j += 1
        return True